attest/verifier/types/src/verification.rs

```diff
@@ -9,15 +9,103 @@ use hex_fmt::{HexFmt, HexList};
 use mc_crypto_digestible::Digestible;
 use mc_crypto_keys::X25519Public;
 use mc_sgx_core_types::QuoteNonce;
+use mc_sgx_dcap_types::{Collateral, Quote3};
 use mc_util_encodings::{Error as EncodingError, FromBase64, FromHex};
 use prost::{
     bytes::{Buf, BufMut},
     encoding::{self, DecodeContext, WireType},
-    DecodeError, Message,
+    DecodeError, Message, Oneof,
 };
 use serde::{Deserialize, Serialize};
 use sha2::{Digest, Sha256};
 
+#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
+pub struct DcapEvidence {
+    pub quote: Option<Quote3<Vec<u8>>>,
+    pub collateral: Option<Collateral>,
+}
+
+const TAG_DCAP_EVIDENCE_QUOTE3: u32 = 1;
+const TAG_DCAP_EVIDENCE_COLLATERAL: u32 = 2;
+
+// Quote3 and Collateral cannot trivially be made to implement prost::Message.
+// Since they implement serde Serialize and Deserialize though, we can manually
+// implement it for DcapEvidence. To do this, we use serde to serialize and
+// deserialize them to/from Vec<u8>
+impl Message for DcapEvidence {
+    fn encode_raw<B>(&self, buf: &mut B)
+    where
+        B: BufMut,
+        Self: Sized,
+    {
+        let quote_bytes: Vec<u8> = mc_util_serial::serialize(&self.quote).unwrap();
+        encoding::bytes::encode(TAG_DCAP_EVIDENCE_QUOTE3, &quote_bytes, buf);
+        let collateral_bytes: Vec<u8> = mc_util_serial::serialize(&self.collateral).unwrap();
```

Comment:
:wrench: we should favor `expect()` over `unwrap()` as it can provide a bit more context in logs
```suggestion
        let quote_bytes: Vec<u8> = mc_util_serial::serialize(&self.quote).expect("Failed to serialize Quote3");
        encoding::bytes::encode(TAG_DCAP_EVIDENCE_QUOTE3, &quote_bytes, buf);
        let collateral_bytes: Vec<u8> = mc_util_serial::serialize(&self.collateral).expect("Failed to serialize Collateral");
```